lib/blocks.php

```diff
@@ -256,22 +257,44 @@ static function() {
 
 		// If the file exists, enqueue it.
 		if ( file_exists( gutenberg_dir_path() . $theme_style_path ) ) {
-
-			if ( file_exists( $stylesheet_path ) ) {
-				// If there is a main stylesheet for this block, append the theme styles to main styles.
-				wp_add_inline_style(
-					"wp-block-{$block_name}",
-					file_get_contents( gutenberg_dir_path() . $theme_style_path )
-				);
+			$theme_style_handle = "wp-block-{$block_name}-theme";
+			if ( wp_style_is( $theme_style_handle, 'registered' ) ) {
+				wp_deregister_style( $theme_style_handle );
+				if ( file_exists( $stylesheet_path ) ) {
+					// If there is a main stylesheet for this block, append the theme styles to main styles.
+					wp_register_style(
+						$theme_style_handle,
+						false,
+						array(),
+						$default_version
+					);
```

Comment:
Shouldn't this be the same as https://github.com/WordPress/gutenberg/pull/52756/files#diff-4167605a44440d5cfb787398fed6a2c0e85f2d83d358d57b9d9dba29ecb50734R284-R287 instead? Why register an empty stylesheet here when the comment above says to append the theme styles to main styles?